### PR TITLE
Sudo mode persistence

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR adds the sudo mode persistence. To do so, we now have a `sudo` object in the database's user, which contains the `sudo_event_ids` array and a `sudo_mode_ttl` key. This key is set with a timestamp in ms corresponding to the expiration date of the "sudo mode" once the user has gone through the send and verify two fa code. 

This flow has been added to the `/account/security/update` route for now. The check to know if user has authorization (is in sudo mode), is done inside the page's `GetServerSideProps`. If he hasn't auth, he is redirected to the sudo page. 

For now, at the end of the `verify-two-fa-validation-code` if it succeeds, the redirection to initially requested page (here : `/account/security/update`) is hardcoded. We will need to rework this eventually. 

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-c0z1kx [feature] - Add validation before changing the password](https://app.clickup.com/t/c0z1kx)

File: src/components/forms/send-two-fa-code-form.tsx [hunk 1]
```diff
@@ -31,22 +34,39 @@ const SendTwoFACodeForm: React.FC<{
...
+            setIsCodeSent(true);
+          }
         });
       }}
     >
-      <p>
-        Chose a contact address below that we’ll send a confirmation code to:
-      </p>
+      {errorMessage ? <WrongInputError>{errorMessage}.</WrongInputError> : null}
+      <p>Chose a contact address below that we’ll send a validation code to:</p>
```
Comment (Fuzznimp) [inline src/components/forms/send-two-fa-code-form.tsx:69]:
```suggestion
      <p>Choose an email address below that we’ll send a validation code to:</p>
```

File: tests/pages/sudo-page.test.tsx [hunk 2]
```diff
@@ -23,9 +23,10 @@ jest.mock("@src/configs/db-client", () => {
 
 const mockedFetchJson = jest.spyOn(fetchJson, "fetchJson");
 
-beforeAll(() => {
-  fetch.resetMocks();
-});
+// const mockedSendTwoFAValidationCode = jest.spyOn(
+//   sendTwoFAValidationCode,
+//   "sendTwoFAValidationCode",
+// );
```
Comment (Fuzznimp) [inline tests/pages/sudo-page.test.tsx:29]: Should this be remove?

Comment (Oluane) [reply tests/pages/sudo-page.test.tsx]: Sure it should 😅

**Review by Fuzznimp — DISMISSED** [review]

**Review by Fuzznimp — DISMISSED** [review]

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**